Add workaround for maintenance mode (#689)
harvester/docs: docs/host/host.md

## Node Maintenance

:::warning

> We found that there is a chance that VMs on a node can encounter an I/O error while the node enters maintenance mode. Please follow this procedure to avoid the problem before a fix is released:

Reviewer: Meanwhile, how does a user recover from this? Would taking the node out of maintenance mode, add the taint, re-enter maintenance mode work?

Author: If a VM enters an I/O error, the only way to recover it is to stop it first and start it again. Let me add this.

Author: Updated in the issue description (somehow I think it's a better place).

> We found that there is a chance that VMs on a node can encounter an I/O error while the node enters maintenance mode. Please follow this procedure to avoid the problem before a fix is released:

Reviewer: ```suggestion
A recent [bug](https://github.com/harvester/harvester/issues/7128) reveals that there is a chance that VMs on a node can encounter an I/O error while in maintenance mode. Users [must|should] follow the following maintenance procedure until a fix is released:
```
I think if this is something that we want every user to follow, we should be more explicit about it.

> A recent [bug](https://github.com/harvester/harvester/issues/7128) reveals that there is a chance that VMs on a node can encounter an I/O error while in maintenance mode. Users should follow the following maintenance procedure until a fix is released:

Reviewer 2: ```suggestion
Due to a recent [bug](https://github.com/harvester/harvester/issues/7128), there is a possibility that virtual machines (VMs) on a node may encounter an I/O error while in maintenance mode. To mitigate this issue until a fix is released, we advise using the following maintenance procedure:
```

> - Set the taint on the maintenance node:

Reviewer 2: ```suggestion
1. Set the taint on the maintenance node:

```

>     ```

Reviewer 2: ```suggestion
    ```sh
```

    kubectl taint node <NODE> --overwrite kubevirt.io/drain=draining:NoSchedule
    ```
- Wait for all VMs to be live-migrated out of the maintenance node.

> - Select **Enable Maintenence Mode** in the GUI.

Reviewer: ```suggestion
- Select **Enable Maintenance Mode** in the GUI.
```

> - Wait for all VMs to be live-migrated out of the maintenance node.

Reviewer 2: ```suggestion

1. Wait for all VMs to be live-migrated out of the maintenance node.
```

> - Select **Enable Maintenance Mode** in the GUI.

Reviewer 2: ```suggestion
1. Select **Enable Maintenance Mode** in the UI.
```

> Once you finish the maintenance work, you need to:

Reviewer 2: ```suggestion
After finishing the maintenance work, you need to:
```

> - Remove the taint on the maintenance node: 

Reviewer 2: ```suggestion
1. Remove the taint on the maintenance node:

```

>     ```

Reviewer 2: ```suggestion
    ```sh
```

    kubectl taint node <NODE> kubevirt.io/drain-
    ```

> - Select **Disable Maintenence Mode** in the GUI.

Reviewer 2: ```suggestion

1. Select **Disable Maintenance Mode** in the UI.
```

> For more information, please check the [issue](https://github.com/harvester/harvester/issues/7128).

Reviewer 2: ```suggestion
For more information, please refer to issue [#7128](https://github.com/harvester/harvester/issues/7128).
```

## Node Maintenance

:::warning

Due to a recent [bug](https://github.com/harvester/harvester/issues/7128), there is a possibility that virtual machines (VMs) on a node may encounter an I/O error while in maintenance mode. To mitigate this issue until a fix is released, we advise using the following maintenance procedure:

1. Set the taint on the maintenance node:

    ```sh
    kubectl taint node <NODE> --overwrite kubevirt.io/drain=draining:NoSchedule
    ```

> 1. Wait for all VMs to be live-migrated out of the maintenance node.

Reviewer 3: Shouldn't the order of these steps be reversed?

```
1. Set the taint on the maintenance node.
1. Select **Enable Maintenance Mode** in the UI.
1. Wait for all VMs to be live-migrated out of the maintenance node.
```

Reviewer 3: @bk201 clarified this with me. All good. 

ref: 
- https://kubevirt.io/user-guide/cluster_admin/node_maintenance/#evict-all-vms-from-a-node
- https://github.com/harvester/harvester/pull/7133
